cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -312,7 +312,7 @@ public RunRecordDetail getRunRecordMeta(ProgramRunId programRunId) throws Except
     if (appMeta == null) {
       throw new ApplicationNotFoundException(programRunId.getParent().getAppReference());
     }
-    RunRecordDetail meta = store.getRun(programRunId);
+    RunRecordDetail meta = store.getRun(programRunId.getReference());
```

Comment:
Whenever a versioned id is passed in as an argument, we need to be treating it as an actual version.

In this case, it looks like the method was improperly implemented earlier, as it is ignoring the version when doing the getLatest() call above. This should be taking a ProgramReference and RunId as input, not a ProgramRunId.

Reply:
It's correct. I changed the order as well: getting the version info from run record and check existence of that application, this way we can fetch the actual version of application